waspc/src/Wasp/AppSpec/Job.hs

```diff
@@ -2,16 +2,35 @@
 
 module Wasp.AppSpec.Job
   ( Job (..),
+    JobExecutor (..),
+    Perform (..),
+    jobPerformOptionsJsonString,
   )
 where
 
 import Data.Data (Data)
 import Wasp.AppSpec.Core.Decl (IsDecl)
 import Wasp.AppSpec.ExtImport (ExtImport)
+import Wasp.AppSpec.JSON (JSON (..))
 
 data Job = Job
-  { perform :: ExtImport
+  { executor :: JobExecutor,
+    perform :: Perform
   }
   deriving (Show, Eq, Data)
 
 instance IsDecl Job
+
+data JobExecutor = Passthrough | PgBoss
+  deriving (Show, Eq, Data)
+
+data Perform = Perform
+  { fn :: ExtImport,
+    options :: Maybe JSON
+  }
+  deriving (Show, Eq, Data)
+
+instance IsDecl Perform
+
+jobPerformOptionsJsonString :: Job -> String
+jobPerformOptionsJsonString job = maybe "{}" (\(JSON str) -> "{" ++ str ++ "}") (options . perform $ job)
```

Comment:
Yup, we could do that! Should we do it now then, do you want to do it?

Reply:
If that is a too big change right now, we can instead, as a solution in the meantime, we could have more general function `WaspJSONtoString :: Wasp.AppSpec.JSON -> String` that does this adding of `{}` so you don't have to implement this very specific `jobPerformOptionsJsonString` function.

Or, maybe implement it as a `Show` instance of `Wasp.AppSpec.JSON`?

Although I don't think it should be a problem to go for the whole thing right now.

Reply:
I think it's a bit large and distant to do now if we tie it into the frontend, as we will also need to signal errors nicely, etc. We could do the simpler `WaspJSONtoString` helper or `show` I suppose.

Reply:
Yeah we can do that then!
I don't think though it would be very hard to modify the Analyzer itself, as JSON is not used in any other place at the moment I think (or maybe just one), so there are not many tests or pieces of code that depend on it.

Reply:
I may save this one for the end and see how long the other ends take to tie up, but can give it a look at the end at least!